sim/src/main/cc/bridges/TracerVModule.cc

```diff
@@ -0,0 +1,242 @@
+// See LICENSE for license details.
+
+#include "BridgeHarness.h"
+#include "TestHarness.h"
+#include "bridges/peek_poke.h"
+#include "bridges/tracerv.h"
+#include "core/bridge_driver.h"
+#include "core/simif.h"
+
+#include <iostream>
+
+static std::vector<bool> get_contiguous(const unsigned bits,
+                                        const unsigned total) {
+  std::vector<bool> ret;
+  for (unsigned i = 0; i < total; i++) {
+    const bool value = (i < bits);
+    ret.emplace_back(value);
+  }
+
+  return ret;
+}
+
+static std::vector<uint64_t> get_iaddrs(const unsigned step,
+                                        const unsigned total) {
+  constexpr uint64_t offset =
+      1024; // should be larger than total but doesn't really matter
+  std::vector<uint64_t> ret;
+
+  for (unsigned i = 0; i < total; i++) {
+    ret.emplace_back(step * offset + i);
+  }
+
+  return ret;
+}
+
+static std::string namei(const unsigned x) {
+  std::stringstream ss;
+  ss << "io_insns_" << x << "_iaddr";
+  return ss.str();
+};
+
+static std::string namev(const unsigned x) {
+  std::stringstream ss;
+  ss << "io_insns_" << x << "_valid";
+  return ss.str();
+};
+
+class TracerVModule : public TestHarness {
```

Comment:
We are not reusing much of the `midasexamples` test harness, but we are adding a cross-project dependency. Please inherit from `simulation_t` directly and duplicate the declaration of the random seed and random number generator, as argument parsing is already done here as well. This test only uses the peek-poke bridge to push in data, we don't care much about inspecting individual pins with the wrappers provided by `TestHarness`.